socs/Lakeshore/Lakeshore370.py

```diff
@@ -0,0 +1,1783 @@
+        Will determine <input/channel> from attributes.
+
+        :param params: RDGRNG parameters
+        :type params: list of str
+
+        :returns: response from ls.msg
+        """
+        assert len(params) == 5
+
+        reply = [str(self.channel_num)]
+        [reply.append(x) for x in params]
+
+        param_str = ','.join(reply)
+        return self.ls.msg(f"RDGRNG {param_str}")
+
+    # Public API
+
+    def get_excitation_mode(self):
+        """Get the excitation mode form RDGRNG?
+
+        :returns: excitation mode, 'current' or 'voltage'
+        :rtype: str
+        """
+        resp = self.get_input_setup()
+        self.mode = mode_key[resp[0]]
+        return self.mode
+
+    def set_excitation_mode(self, excitation_mode):
+        """Set the excitation mode to either voltage excitation or current
+        exitation.
+
+        :param excitation_mode: mode we want, must be 'current' or 'voltage'
+        :type excitation_mode: str
+
+        :returns: reply from RDGRNG call
+        :rtype: str
+
+        """
+        assert excitation_mode in ['voltage', 'current']
+
+        resp = self.get_input_setup()
+        resp[0] = mode_lock[excitation_mode]
+
+        self.mode = mode_key[resp[0]]
+
+        return self._set_input_setup(resp)
+
+    def get_excitation(self):
+        """Get excitation value from RDGRNG?
+
+        :returns: excitation value in volts or amps, depending on mode
+        :rtype: float
+        """
+        resp = self.get_input_setup()
+        _mode = resp[0]
+        _excitation = resp[1]
+
+        excitation_key = {'0': voltage_excitation_key,
+                            '1': current_excitation_key}
+
+        self.excitation = excitation_key[_mode][int(_excitation)]
+
+        return self.excitation
+
+    def set_excitation(self, excitation_value):
+        """Set voltage/current exitation to specified value via RDGRNG command.
+
+        :param excitation_value: value in volts/amps of excitation
+        :type excitation_value: float
+
+        :returns: response from RDGRNG command
+        :rtype: str
+        """
+        _mode = self.mode
+
+        if _mode == 'voltage':
+            excitation_lock = voltage_excitation_lock
+        elif _mode == 'current':
+            excitation_lock = current_excitation_lock
+
+        closest_value = min(excitation_lock, key=lambda x: abs(x-excitation_value))
+
+        resp = self.get_input_setup()
+        resp[1] = str(excitation_lock[closest_value])
+
+        return self._set_input_setup(resp)
+
+    def enable_autorange(self):
+        """Enable auto range for channel via RDGRNG command."""
+        resp = self.get_input_setup()
+        #order of resp args switch for range, autorange in LS370
+        resp[3] = '1'     
+
+        #all LS370 channels respond to this command
+        for c in self.ls.channels: 
+            c.autorange = autorange_key[resp[3]]
+
+        #TODO: move method to LS370 class, fix references in agent
```

Comment:
Oh. It's because I've observed that calling this method on any one channel applies it (in the machine) to all channels. So I cludged this with the for loop, which loops over all channels. But given this behavior, this method is really more of an LS370 method, not a channel-specific method.

Reply:
Hmm, those options should be individually configurable. Looking at the 370 manual, they will be globally configured if the channel specified is channel "0", else the `RDGRNG` command will assign properties per channel. Maybe a channel_num zero was initialized somehow and causing global changes to occur.